### ISSUE TITLE
Invalid violation with the arrow-parens rule when using generics on a function

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#1442]
### Bug Report

- __TSLint version__: `3.14.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: Visual Studio 2015 + `gulp-tslint` 6.0.2

#### TypeScript code being linted

```ts
//OK
public foo = (method: () => string) => {
    method();
};

//Violation??
public bar = <T>(method: () => T) => {
    method();
};
```

with `tslint.json` configuration:

```json
"arrow-parens": true,
```

#### Actual behavior

The `bar` method above throws a violation `Parentheses are required around the parameters of an arrow function definition`.  This seems incorrect to me... unless I'm missing something obvious here.

#### Expected behavior

No violations

**Comment by JKillian** [comment]
cc @YuichiNukiyama

**Comment by YuichiNukiyama** [comment]
Oops, this is my mistaken... I'll send PR.
